web/src/main/java/com/zebrunner/reporting/web/TestRunController.java

```diff
@@ -161,6 +164,25 @@ public TestRunType getTestRun(@PathVariable("id") long id) {
         return testRunService.search(sc, projectNames, filterId);
     }
 
+    @PostMapping("{id}/artifacts")
+    @Override
+    public Set<TestRunArtifactDTO> attachArtifacts(@PathVariable("id") Long id,
+                                                   @RequestBody @Valid Set<TestRunArtifactDTO> testRunArtifactDTOS) {
+        Set<TestRunArtifact> testRunArtifacts = testRunArtifactDTOS.stream()
+                                                                .map(testRunArtifact -> mapper.map(testRunArtifact, TestRunArtifact.class))
```

Comment:
Super minor note: use shorter name in lambdas here - it is usually clear from context what kind of object is manipulated here. I'd use `artifact`. Consider this my personal preference, feel free to ignore. Same applies to line 176 in this file